glider_kinematics_gazebo_plugins/kinematics_plugins/include/kinematics_plugins/KinematicsPlugin.hh

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2016 The UUV Simulator Authors.
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// \file kinematics_plugins.hh
+/// \brief Plugin that for the vehicle kinematics
+
+#ifndef __KINEMATICS_PLUGIN_HH__
+#define __KINEMATICS_PLUGIN_HH__
+
+#include <map>
+#include <cmath>
+#include <string>
+
+#include <gazebo/gazebo.hh>
+#include <gazebo/transport/TransportTypes.hh>
+#include <kinematics_plugins/GaussMarkovProcess.hh>
+#include <sdf/sdf.hh>
+
+namespace gazebo
+{
+  /// \brief Class for the vehicle kinematics
+  class KinematicsPlugin : public ModelPlugin
+  {
+    /// \brief Class constructor
+    public: KinematicsPlugin();
+
+    /// \brief Class destructor
+    public: virtual ~KinematicsPlugin();
+
+    // Documentation inherited.
+    public: virtual void Load(physics::ModelPtr _model,
+        sdf::ElementPtr _sdf);
+
+    // Documentation inherited.
+    public: virtual void Init();
+
+    /// \brief Update the simulation state.
+    /// \param[in] _info Information used in the update event.
+    public: void Update(const common::UpdateInfo &_info);
+
+    /// \brief Publish current velocity and the pose of its frame
+    protected: void PublishCurrentVelocity();
+
+    /// \brief Update event
+    protected: event::ConnectionPtr updateConnection;
+
+    /// \brief Pointer to the model
+    protected: physics::ModelPtr model;
+
+    /// \brief Pointer to sdf
+    protected: sdf::ElementPtr sdf;
+
+    /// \brief True if the sea surface is present
+    protected: bool hasSurface;
+
+    /// \brief Pointer to a node for communication
+    protected: transport::NodePtr node;
+
+    /// \brief Map of publishers
+    protected: std::map<std::string, transport::PublisherPtr>
+      publishers;
+
+    /// \brief Current velocity topic
+    protected: std::string currentVelocityTopic;
+
+    /// \brief Namespace for topics and services
+    protected: std::string ns;
+
+    /// \brief Gauss-Markov process instance for the current velocity
```

Comment:
If I understand this correctly, the plugin adds uncertainty (noise) to the state commands.  This is a good idea, but I don't think we need it.  For the purposes of this use case, we can set the states (velocities) exactly.  The simulated sensor measurements will add uncertainty to the observations for the purposes of exercising the algorithms.